streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java

```diff
@@ -1267,8 +1268,14 @@ public void cancel(Application appParam) throws Exception {
                 savePoint.setAppId(application.getId());
                 savePoint.setLatest(true);
                 savePoint.setType(CheckPointType.SAVEPOINT.get());
-                savePoint.setTriggerTime(now);
+                savePoint.setTriggerTime(triggerTime);
                 savePoint.setCreateTime(now);
+                savePoint.setJobId(cancelRequest.jobId());
+                savePoint.setEndTime(now);
+                savePoint.setClusterId(application.getFlinkClusterId());
+                if (appParam.getExecutionMode() != null) {
+                  savePoint.setExecutionMode(ExecutionMode.of(appParam.getExecutionMode()).name());
```

Comment:
`ExecutionMode.of(appParam.getExecutionMode()).name()` == `appParam.getExecutionMode()`, right?

If yes, how about `savePoint.setExecutionMode(appParam.getExecutionMode())`?

And the `if (appParam.getExecutionMode() != null) ` can be removed.

Reply:
good catch

Reply:
`appParam.getExecutionMode()` is also string, right?